src/main/java/org/commonjava/util/partyline/FileTree.java

```diff
@@ -238,7 +242,17 @@ private boolean unlockAssociatedEntries( final FileEntry entry, final String lab
         while ( alsoLocked != null )
         {
             logger.trace( "ALSO Unlocking: {}", alsoLocked.name );
-            alsoLocked.lock.unlock( label );
+            UnlockStatus unlockStatus = alsoLocked.lock.unlock( label );
+            try
+            {
+                filesystem.updateDominantLocks( entry.file.getPath(), unlockStatus );
+            }
+            catch ( IOException e )
```

Comment:
I think we should throw this IOException, then throw it again from FileTree.unlock(). In FileTree.clearLocks() we'll need to catch and log it, though, since clearLocks is exception/finally handling logic.